doc/nrf/ug_matter_overview_architecture_integration.rst

```diff
@@ -91,3 +91,25 @@ For this design, the Wi-Fi driver on the application core communicates with the
    :alt: Multiprotocol Wi-Fi and Bluetooth LE architecture (nRF53 with the nRF7002 Wi-Fi 6 Companion IC)
 
    Multiprotocol Wi-Fi and Bluetooth LE architecture (nRF53 with the nRF7002 Wi-Fi 6 Companion IC)
+
+Switchable Matter over Wi-Fi and Matter over Thread
+===================================================
+
+In this design, the Matter stack, the OpenThread stack, Wi-Fi stack, and the Bluetooth LE stack run on a single SoC.
+The OpenThread stack and the Wi-Fi stack are placed in the separate partitions of the external flash.
```

Comment:
That's not entirely true. It's not like we have one common application and only protocol stacks are put into separate external flash. We have two separate variants of this application - the first one is application working with Matter over Thread and the second one is application working with Matter over Wi-Fi. We run one application variant from the FLASH memory at a time, while we also keep both variants copies in the external flash. In case of switching we read the whole new variant from external flash and put it into internal flash.

Reply:
Rephrased.